packages/blocks/src/api/parser.js

```diff
@@ -456,6 +441,34 @@ export function createBlockWithFallback( blockNode ) {
 		}
 		name = 'core/embed';
 	}
+	return { name, attributes };
+}
+
+/**
+ * Creates a block with fallback to the unknown type handler.
+ *
+ * @param {Object} blockNode Parsed block node.
+ *
+ * @return {?Object} An initialized block object (if possible).
+ */
+export function createBlockWithFallback( blockNode ) {
+	const { blockName: originalName } = blockNode;
+	let { attrs: attributes, innerBlocks = [], innerHTML } = blockNode;
+	const { innerContent } = blockNode;
+	const freeformContentFallbackBlock = getFreeformContentHandlerName();
+	const unregisteredFallbackBlock =
+		getUnregisteredTypeHandlerName() || freeformContentFallbackBlock;
+
+	attributes = attributes || {};
+
+	// Trim content to avoid creation of intermediary freeform segments.
+	innerHTML = innerHTML.trim();
+
+	// Use type from block content if available. Otherwise, default to the
+	// freeform content fallback.
+	let name = originalName || freeformContentFallbackBlock;
+
+	name = convertDerivativeBlocks( name, attributes ).name;
```

Comment:
Looking at this line, one would expect no side effects to happen. However, `attributes` is being mutated behind the scenes. We should fix this by either:

* Eliminating side effects and making `convertDerivativeBlocks` return `{ name, attributes }`
* Making it very clear that there are side effects and eliminating return values: `(void) convertDerivativeBlocks( blockNode )`

The former would be the least surprising approach.

Reply:
Yes I thought about making this one pure, but didn't believe it was so much needed as it actually conditionally adds some attributes that wouldn't exist in the first place. In other cases both name and attributes are returned as are.

I can make it pure of course :)